File: solutions/0053-maximum-subarray/solution.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        max_final=nums[0]
        max_prev=0
        for i in range (len(nums)):
            if max_prev<0:
                max_prev=0
                
            max_prev+=nums[i]

            if max_prev>max_final:
                max_final=max_prev
        
        return max_final

File: solutions/0053-maximum-subarray/solution.c
int maxSubArray(int* nums, int numsSize) {
    int max_num=nums[0];
    int max_prev=0;
    for(int i=0;i<numsSize;i++){
        if(max_prev<0){
            max_prev=0;
        }
        max_prev+=nums[i];
        if(max_prev>max_num){
            max_num=max_prev;
        }
    }
    return max_num;
}
